### DISCUSSION TITLE
Don't need PaymentIntentPersistence

### DISCUSSION THREAD

**Discussion by infofromca** [OrchardCMS/OrchardCore.Commerce#498] [Ideas]
It will have the same  issue as #475 in future. So, should we delete it?

**Comment by sarahelsaig** [comment]
No, we need some kind of persistence. As I [mentioned over there](https://github.com/OrchardCMS/OrchardCore.Commerce/issues/475#:~:text=Web%20developers%20using%20OCC%20can%20implement%20their%20own%20IShoppingCartPersistence%20which%20uses%20site%20setting%2C%20an%20external%20database%2C%20etc%20for%20persistent%20storage.), you can replace the default implementation with a custom one that uses persistent storage (custom table, site setting, etc)  instead of volatile session storage. If you have a suggestion for an alternative implementation, we can discuss it here. Later we can make it a site setting to pick persistence type.
